www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,425 @@
+  grade?: number;
+  gradeUnit?: string;
+  tonnageUnit?: string;
+}
+
+export const NewMineralSiteModal: React.FC<NewMineralSiteModalProps> = ({ commodity, visible, onClose }) => {
+  const { mineralSiteStore, dedupMineralSiteStore, userStore, commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore } = useStores();
+  const [form] = Form.useForm();
+  const [recordId, setRecordId] = useState("");
+  const [selectedSourceType, setSelectedSourceType] = useState<string | null>(null);
+
+  const commodityOptions = useMemo(() => {
+    return commodityStore.list.map((comm) => ({
+      value: comm.uri,
+      label: comm.name,
+    }));
+  }, [commodityStore.records]);
+  const currentUserUrl = userStore.getCurrentUser()!.url;
+  const countryOptions = useMemo(() => {
+    return countryStore.list.map((country) => ({
+      value: country.uri,
+      label: country.name,
+    }));
+  }, [countryStore.records]);
+
+  const stateOptions = useMemo(() => {
+    return stateOrProvinceStore.list.map((state) => ({
+      value: state.id,
+      label: state.name,
+    }));
+  }, [stateOrProvinceStore.records]);
+
+  const depositTypeOptions = useMemo(() => {
+    return depositTypeStore.list.map((type) => ({
+      value: type.id,
+      label: type.name,
+    }));
+  }, [depositTypeStore.records]);
+
+  const unitOptions = useMemo(() => {
+    return unitStore.list.map((unit) => ({
+      value: unit.id,
+      label: unit.name,
+    }));
+  }, [unitStore.records]);
+
+  const handleSourceTypeChange = (e: RadioChangeEvent) => {
+    const value = e.target.value;
+    setSelectedSourceType(value);
+
+    const refDocUrl = form.getFieldValue("refDoc");
+    if (value === "unpublished") {
+      form.setFieldsValue({ sourceId: `unpublished::${currentUserUrl}` });
+    } else if (refDocUrl) {
+      form.setFieldsValue({ sourceId: `${value}::${refDocUrl}` });
+    }
+  };
+
+
+
+  const handleSave = async (values: FormValues) => {
+    try {
+      let location = undefined;
+      if (values.latitude !== undefined && values.longitude !== undefined) {
+        location = `POINT (${values.longitude} ${values.latitude})`;
+      }
+
+      const countries = values.country
+        ? [
+          new CandidateEntity({
+            observedName: countryStore.getByURI(values.country)!.name,
+            source: currentUserUrl,
+            normalizedURI: values.country,
+            confidence: 1.0,
+          }),
+        ]
+        : [];
+
+      const statesOrProvinces = values.stateOrProvince
+        ? [
+          new CandidateEntity({
+            observedName: stateOrProvinceStore.getByURI(values.stateOrProvince)!.name,
+            source: currentUserUrl,
+            normalizedURI: values.stateOrProvince,
+            confidence: 1.0,
+          }),
+        ]
+        : [];
+
+      const commodity1 = commodity.id;
+      const referenceDocument = new Document({
+        uri: values.refDoc,
+        title: values.refDoc,
```

Comment:
refDoc is URL then it's not the title

Reply:
I am not sure what the title should be. Can you please help me with it ?